router/core/access_controller.go

```diff
@@ -1,12 +1,29 @@
 package core
 
 import (
+	"crypto/subtle"
 	"errors"
+	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/wundergraph/cosmo/router/pkg/authentication"
 )
 
+// IntrospectionAuthMode defines how introspection queries are authenticated.
+type IntrospectionAuthMode string
+
+const (
+	// IntrospectionAuthModeFull requires normal authentication for introspection queries.
+	IntrospectionAuthModeFull IntrospectionAuthMode = "full"
+
+	// IntrospectionAuthModeToken requires a specific introspection token for introspection queries.
+	IntrospectionAuthModeToken IntrospectionAuthMode = "token"
+
+	// IntrospectionAuthModeSkip bypasses authentication for introspection queries.
+	IntrospectionAuthModeSkip IntrospectionAuthMode = "skip"
+)
```

Comment:
It's good practice with the Go string enums like this to have a named default like

```go
const IntrospectionAuthModeInvalid IntrospectionAuthMode = ""
```

Reply:
fixed